app/Jobs/Brevo/ProcessBrevoWebhook.php

```diff
@@ -0,0 +1,475 @@
+        }
+    }
+
+    private function discoverInvitation(string $message_id)
+    {
+        $invitation = false;
+
+        if ($invitation = InvoiceInvitation::where('message_id', $message_id)->first()) {
+            $this->entity = 'invoice';
+            return $invitation;
+        } elseif ($invitation = QuoteInvitation::where('message_id', $message_id)->first()) {
+            $this->entity = 'quote';
+            return $invitation;
+        } elseif ($invitation = RecurringInvoiceInvitation::where('message_id', $message_id)->first()) {
+            $this->entity = 'recurring_invoice';
+            return $invitation;
+        } elseif ($invitation = CreditInvitation::where('message_id', $message_id)->first()) {
+            $this->entity = 'credit';
+            return $invitation;
+        } elseif ($invitation = PurchaseOrderInvitation::where('message_id', $message_id)->first()) {
+            $this->entity = 'purchase_order';
+            return $invitation;
+        } else {
+            return $invitation;
+        }
+    }
+
+    public function getRawMessage(string $message_id)
+    {
+
+        $Brevo = new TransactionalEmailsApi(null, Configuration::getDefaultConfiguration()->setApiKey('api-key', config('services.brevo.key')));
+        $messageDetail = $Brevo->getTransacEmailContent($message_id);
+        return $messageDetail;
+
+    }
+
+
+    public function getBounceId(string $message_id): ?int
+    {
+
+        $messageDetail = $this->getRawMessage($message_id);
+
+
+        $event = collect($messageDetail->getEvents())->first(function ($event) {
+
+            return $event?->Details?->BounceID ?? false;
+
+        });
+
+        return $event?->Details?->BounceID ?? null;
+
+    }
+
+    // TODO
+    private function fetchMessage(): array
+    {
+        if (strlen($this->request['message-id']) < 1) {
+            return $this->default_response;
+        }
+
+        try {
+
+            $messageDetail = $this->getRawMessage($this->request['message-id']);
+
+            $recipients = $this->request["email"];
+            $subject = $messageDetail->getSubject() ?? '';
+
+            $events = collect($messageDetail->getEvents())->map(function (GetTransacEmailContentEvents $event) { // @turbo124 event does only contain name & time property, how to handle transformation?!
+
+                return [
+                    'bounce_id' => $event?->Details?->BounceID ?? '',
+                    'recipient' => $event->Recipient ?? '',
+                    'status' => $event->Type ?? '',
+                    'delivery_message' => $event->Details->DeliveryMessage ?? $event->Details->Summary ?? '',
+                    'server' => $event->Details->DestinationServer ?? '',
+                    'server_ip' => $event->Details->DestinationIP ?? '',
+                    'date' => \Carbon\Carbon::parse($event->getTime())->format('Y-m-d H:i:s') ?? '',
```

Comment:
@turbo124 this results in all cases for the history in the string, which may be incorrect

Reply:
sorry pushed to late, I mean the property: 'delivery_message' (please look at latest file)

Reply:
delivery message is typically the smtp response from the server, so typically for the open / spam events, it would be the same as the initial one.